undefined method `binary_path' for nil:NilClass (NoMethodError)

## System

- OS: Mac OS 10.13.5
- `ruby -v`: ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin16]
- `bundle -v`: Bundler version 1.16.4
- `bundle info arduino_ci`:
```
  * arduino_ci (0.1.16)
	Summary: Tools for building and unit testing Arduino libraries
	Homepage: http://github.com/ianfixes/arduino_ci
	Path: /Users/scls/.rvm/gems/ruby-2.3.0/gems/arduino_ci-0.1.16
```
- `g++ -v`:
```
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.11.45.5)
Target: x86_64-apple-darwin17.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```
- Arduino IDE version: 1.8.6

- `java -version`:
```
java version "1.8.0_192"
Java(TM) SE Runtime Environment (build 1.8.0_192-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.192-b12, mixed mode)
```

##  Issue
I'm just trying to run unit test locally on my Mac...
(I'm not trying CI for now)

```
$ bundle exec arduino_ci_remote.rb
Arduino package seems to have been downloaded already
Extracting archive with Zip.....................................................................................................
Arduino package seems to have been installed already
Located Arduino binary... /Users/scls/.rvm/gems/ruby-2.3.0/gems/arduino_ci-0.1.16/exe/arduino_ci_remote.rb:311:in `block in <top (required)>': undefined method `binary_path' for nil:NilClass (NoMethodError)
	from /Users/scls/.rvm/gems/ruby-2.3.0/gems/arduino_ci-0.1.16/exe/arduino_ci_remote.rb:40:in `perform_action'
	from /Users/scls/.rvm/gems/ruby-2.3.0/gems/arduino_ci-0.1.16/exe/arduino_ci_remote.rb:75:in `inform'
	from /Users/scls/.rvm/gems/ruby-2.3.0/gems/arduino_ci-0.1.16/exe/arduino_ci_remote.rb:311:in `<top (required)>'
	from /Users/scls/.rvm/gems/ruby-2.3.0/bin/arduino_ci_remote.rb:23:in `load'
	from /Users/scls/.rvm/gems/ruby-2.3.0/bin/arduino_ci_remote.rb:23:in `<main>'
	from /Users/scls/.rvm/gems/ruby-2.3.0/bin/ruby_executable_hooks:15:in `eval'
	from /Users/scls/.rvm/gems/ruby-2.3.0/bin/ruby_executable_hooks:15:in `<main>'
```

Any idea?

I would like to run unit test for Arduino Mega 2560 but I'm not sure what should be in `.arduino-ci.yml`

I've tried `mega` and also `mega2560`

with `mega`
```
10:39 $ cat .arduino-ci.yml
unittest:
  platforms:
    - mega

compile:
  platforms:
    - mega
✔ ~/github/scls19fr/gokart_multilaptimer [master|✚ 1…3]
10:39 $ bundle exec arduino_ci_remote.rb
Located Arduino binary...                                /Users/scls/Arduino.app
The set of compilers (1) isn't empty...                                        ✓
Checking g++ version...
    Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
    Apple LLVM version 10.0.0 (clang-1000.11.45.5)
    Target: x86_64-apple-darwin17.7.0
    Thread model: posix
    InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
...Checking g++ version                                                        ✓
libasan availability for g++...                                             true
Requested unittest platform 'mega' is defined in 'platforms' YML...            ✗
This may indicate a problem with ArduinoCI, or your configuration
Failures: 1
Last message:  $  java -cp /Users/scls/Arduino.app/Contents/Java/* -DAPP_DIR=/Users/scls/Arduino.app/Contents/Java -Dfile.encoding=UTF-8 -Dapple.awt.UIElement=true -Xms128M -Xmx512M processing.app.Base --get-pref
========== Stdout:
Bibliothèque non valide trouvée dans /Users/scls/Documents/Arduino/libraries/gokart_multilaptimer : /Users/scls/Documents/Arduino/libraries/gokart_multilaptimer
Bibliothèque non valide trouvée dans /Users/scls/Documents/Arduino/libraries/gokart_multilaptimer : /Users/scls/Documents/Arduino/libraries/gokart_multilaptimer
#PREFDUMP#
last.sketch.count=1
update.check=true
platform.auto_file_type_associations=true
gui.scale=auto
editor.window.width.default=500
editor.window.height.default=600
editor.window.width.min=400
editor.window.height.min=290
editor.code_folding=false
editor.font=Monaco,plain,12
editor.antialias=true
editor.external=false
editor.caret.blink=true
editor.invalid=false
editor.linenumbers=false
editor.keys.alternative_cut_copy_paste=false
editor.keys.shift_backspace_is_delete=true
editor.keys.home_and_end_beginning_end_of_doc=false
console=true
console.output.file=stdout.txt
console.error.file=stderr.txt
console.lines=4
console.auto_clear=true
console.length=500
editor.tabs.expand=true
editor.tabs.size=2
editor.indent=true
editor.auto_close_braces=true
editor.divider.size=0
editor.save_on_verify=true
run.options=
run.options.memory=false
run.options.memory.initial=64
run.options.memory.maximum=256
run.display=1
export.application.platform=true
export.application.fullscreen=false
export.application.stop=true
export.applet.separate_jar_files=false
export.delete_target_folder=true
preproc.save_build_files=false
preproc.color_datatype=true
preproc.web_colors=true
preproc.enhanced_casting=true
preproc.substitute_floats=true
preproc.substitute_unicode=true
preproc.output_parse_tree=false
preproc.imports.list=java.applet.*,java.awt.Dimension,java.awt.Frame,java.awt.event.MouseEvent,java.awt.event.KeyEvent,java.awt.event.FocusEvent,java.awt.Image,java.io.*,java.net.*,java.text.*,java.util.*,java.util.zip.*,java.util.regex.*
run.present.bgcolor=#666666
run.present.stop.color=#cccccc
run.present.exclusive=true
target_package=arduino
target_platform=avr
board=mega
software=ARDUINO
build.warn_data_percentage=75
programmer=arduino:arduinoisporg
upload.using=bootloader
upload.verify=true
serial.port=/dev/cu.usbmodemFD121
serial.databits=8
serial.stopbits=1
serial.parity=N
serial.debug_rate=9600
serial.line_ending=1
editor.languages.current=
preferences.readonly=false
runtime.ide.path=/Users/scls/Arduino.app/Contents/Java
runtime.ide.version=10806
custom_cpu=mega_atmega2560
last.ide.1.8.2.daterun=1536221650
last.ide.1.8.2.hardwarepath=/Applications/Arduino.app/Contents/Java/hardware
last.ide.1.8.6.daterun=1547285993
last.ide.1.8.6.hardwarepath=/Users/scls/Arduino.app/Contents/Java/hardware
last.screen.height=900
last.screen.width=1440
last.serial.location=591,23,769,831
last.sketch.default.location=92,36,971,770,554,0
last.sketch.default.path=/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/multilaptimer.ino
last.sketch0.location=92,36,971,770,554,0
last.sketch0.path=/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/multilaptimer.ino
last.sketch1.location=161,67,500,600,456,0
last.sketch1.path=/Users/scls/github/scls19fr/ekart_daq_2018_2019/tuto_05/tuto_05_stand/tuto_05_stand.ino
last.sketch2.location=470,150,500,600,456,0
last.sketch2.path=/Users/scls/github/scls19fr/ekart_daq_2018_2019/tuto_05/tuto_05_stand/tuto_05_stand.ino
last.sketch3.location=862,408,998,744,579,0
last.sketch3.path=/Users/scls/github/scls/gokart_logger/gokart_logger/gokart_logger.ino
last.sketch4.location=350,480,669,763,598,0
last.sketch4.path=/Users/scls/github/scls/gokart_logger/clock_set_2/clock_set_2.ino
last.sketch5.location=623,23,715,830,665,0
last.sketch5.path=/Users/scls/Documents/Arduino/clock_set/clock_set.ino
recent.sketches=/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/multilaptimer.ino,/Users/scls/Downloads/Nokia5110Menu/Nokia5110Menu.ino,/Users/scls/github/scls19fr/ekart_daq_2018_2019/tuto_05/tuto_05_kart/tuto_05_kart.ino,/Users/scls/github/scls19fr/ekart_daq_2018_2019/tuto_05/tuto_05_stand/tuto_05_stand.ino,/Users/scls/github/scls/gokart_logger/RPMmeter_lcd/RPMmeter_lcd.ino,/Users/scls/github/scls/gokart_logger/RPMmeter/RPMmeter.ino,/Users/scls/github/scls/gokart_logger/RPMmeter_serial/RPMmeter_serial.ino,/Users/scls/github/scls/gokart_logger/clock_set_3/clock_set_3.ino,/Users/scls/Documents/Arduino/clock_set/clock_set.ino,/Users/scls/github/scls/gokart_logger/gokart_logger/gokart_logger.ino,/Users/scls/github/scls/gokart_logger/clock_set/clock_set.ino,/Users/scls/github/scls/gokart_logger/clock_set_2/clock_set_2.ino,/Users/scls/Documents/Arduino/clock_set_2/clock_set_2.ino,/Users/scls/github/scls/gokart_logger/lcd_i2c_nav_keypad/lcd_i2c_nav_keypad.ino,/Users/scls/github/scls/gokart_logger/RPMmeter/src/RPMmeter/RPMmeter.ino,/Users/scls/Documents/Arduino/lcd_i2c_nav_keypad/lcd_i2c_nav_keypad.ino,/Users/scls/github/scls/arduino-fsm/examples/light_switch/light_switch.ino,/Users/scls/Documents/Arduino/temp/temp.ino,/Users/scls/Documents/Arduino/libraries/arduino-menusystem/examples/lcd_nav/lcd_nav.ino,/Users/scls/Documents/Arduino/libraries/arduino-menusystem/examples/lcd_nav_keypad/lcd_nav_keypad.ino,/Users/scls/Documents/Arduino/libraries/arduino-menusystem/examples/lcd_i2c_nav_keypad/lcd_i2c_nav_keypad.ino,/Users/scls/Documents/Arduino/lcd_i2c_nav/lcd_i2c_nav.ino,/Users/scls/Documents/Arduino/test_menu/test_menu.ino,/Users/scls/Documents/Arduino/menu/menu.ino,/Users/scls/Documents/Arduino/lcd/lcd.ino,/Users/scls/Documents/Arduino/serial/serial.ino,/Users/scls/github/scls/gokart_logger/chrono/chrono.ino,/Users/scls/Documents/Arduino/libraries/SdFat/examples/SdFormatter/SdFormatter.ino,/Users/scls/Documents/Arduino/libraries/SdFat/examples/ReadWrite/ReadWrite.ino,/Users/scls/Documents/Arduino/chrono/chrono.ino,/Users/scls/Documents/Arduino/RPMmeter/RPMmeter.ino,/Users/scls/Documents/Arduino/RPMmeter_serial/RPMmeter_serial.ino,/Users/scls/Downloads/Fuelino-SW-1.0-beta5-master/Fuelino-SW-1.0-beta5/libraries/SDFatYield/examples/dataLogger/dataLogger.ino,/Users/scls/Documents/Arduino/HelloWorld2/HelloWorld2.ino,/Users/scls/Documents/Arduino/HelloWorld/HelloWorld.ino,/Users/scls/Downloads/gokart_logger-master/gokart_logger/gokart_logger.ino,/Users/scls/Documents/Arduino/libraries/Adafruit_LiquidCrystal/examples/HelloWorld_SPI/HelloWorld_SPI.pde,/Users/scls/Documents/Arduino/libraries/Adafruit_LiquidCrystal/examples/HelloWorld_i2c/HelloWorld_i2c.pde,/Users/scls/Documents/Arduino/libraries/Adafruit_ADXL345/examples/sensortest/sensortest.pde,/Users/scls/github/scls/8652a41d3ef3b6027c74cf0027b1d8dd_gokart_arduino/gokart_logger/gokart_logger.ino,/Users/scls/Documents/Arduino/sample_serial/sample_serial.ino
serial.port.file=cu.usbmodemFD121
serial.port.iserial=75633313133351D082B0
sketchbook.path=/Users/scls/Documents/Arduino
update.id=906145261407501964
runtime.os=macosx
runtime.tools.openocd-0.9.0+0.1.path=/Users/scls/Library/Arduino15/packages/Intel/tools/openocd/0.9.0+0.1
runtime.tools.Intel-openocd-0.9.0+0.1.path=/Users/scls/Library/Arduino15/packages/Intel/tools/openocd/0.9.0+0.1
runtime.tools.openocd.path=/Users/scls/Library/Arduino15/packages/Intel/tools/openocd/0.9.0+0.1
runtime.tools.arduinoOTA-1.1.1.path=/Users/scls/Library/Arduino15/packages/arduino/tools/arduinoOTA/1.1.1
runtime.tools.arduino-arduinoOTA-1.1.1.path=/Users/scls/Library/Arduino15/packages/arduino/tools/arduinoOTA/1.1.1
runtime.tools.arduinoOTA.path=/Users/scls/Library/Arduino15/packages/arduino/tools/arduinoOTA/1.2.1
runtime.tools.avrdude-6.3.0-arduino9.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avrdude/6.3.0-arduino9
runtime.tools.arduino-avrdude-6.3.0-arduino9.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avrdude/6.3.0-arduino9
runtime.tools.avrdude.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avrdude/6.3.0-arduino14
runtime.tools.arduino101load-2.0.1.path=/Users/scls/Library/Arduino15/packages/Intel/tools/arduino101load/2.0.1
runtime.tools.Intel-arduino101load-2.0.1.path=/Users/scls/Library/Arduino15/packages/Intel/tools/arduino101load/2.0.1
runtime.tools.arduino101load.path=/Users/scls/Library/Arduino15/packages/Intel/tools/arduino101load/2.0.1
runtime.tools.avrdude-6.3.0-arduino14.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avrdude/6.3.0-arduino14
runtime.tools.arduino-avrdude-6.3.0-arduino14.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avrdude/6.3.0-arduino14
runtime.tools.arduinoOTA-1.2.1.path=/Users/scls/Library/Arduino15/packages/arduino/tools/arduinoOTA/1.2.1
runtime.tools.arduino-arduinoOTA-1.2.1.path=/Users/scls/Library/Arduino15/packages/arduino/tools/arduinoOTA/1.2.1
runtime.tools.dfu-util-0.9.0-arduino1.path=/Users/scls/Library/Arduino15/packages/arduino/tools/dfu-util/0.9.0-arduino1
runtime.tools.arduino-dfu-util-0.9.0-arduino1.path=/Users/scls/Library/Arduino15/packages/arduino/tools/dfu-util/0.9.0-arduino1
runtime.tools.dfu-util.path=/Users/scls/Library/Arduino15/packages/arduino/tools/dfu-util/0.9.0-arduino1
runtime.tools.avr-gcc-4.9.2-atmel3.5.4-arduino2.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avr-gcc/4.9.2-atmel3.5.4-arduino2
runtime.tools.arduino-avr-gcc-4.9.2-atmel3.5.4-arduino2.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avr-gcc/4.9.2-atmel3.5.4-arduino2
runtime.tools.avr-gcc.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avr-gcc/5.4.0-atmel3.6.1-arduino2
runtime.tools.arc-elf32-1.6.9+1.0.1.path=/Users/scls/Library/Arduino15/packages/Intel/tools/arc-elf32/1.6.9+1.0.1
runtime.tools.Intel-arc-elf32-1.6.9+1.0.1.path=/Users/scls/Library/Arduino15/packages/Intel/tools/arc-elf32/1.6.9+1.0.1
runtime.tools.arc-elf32.path=/Users/scls/Library/Arduino15/packages/Intel/tools/arc-elf32/1.6.9+1.0.1
runtime.tools.avr-gcc-5.4.0-atmel3.6.1-arduino2.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avr-gcc/5.4.0-atmel3.6.1-arduino2
runtime.tools.arduino-avr-gcc-5.4.0-atmel3.6.1-arduino2.path=/Users/scls/Library/Arduino15/packages/arduino/tools/avr-gcc/5.4.0-atmel3.6.1-arduino2
runtime.tools.flashpack-2.0.0.path=/Users/scls/Library/Arduino15/packages/Intel/tools/flashpack/2.0.0
runtime.tools.Intel-flashpack-2.0.0.path=/Users/scls/Library/Arduino15/packages/Intel/tools/flashpack/2.0.0
runtime.tools.flashpack.path=/Users/scls/Library/Arduino15/packages/Intel/tools/flashpack/2.0.0
runtime.platform.path=/Users/scls/Arduino.app/Contents/Java/hardware/arduino/avr
runtime.hardware.path=/Users/scls/Arduino.app/Contents/Java/hardware/arduino
========== Stderr:
Chargement de la configuration
Initialisation des paquets...
Préparation des cartes
```

What is done is unclear here
Is it failing? is it passing?

with `mega2560`

```
✘-1 ~/github/scls19fr/gokart_multilaptimer [master|✚ 1…3]
10:39 $ open .arduino-ci.yml
✔ ~/github/scls19fr/gokart_multilaptimer [master|✚ 1…3]
10:40 $ cat .arduino-ci.yml
unittest:
  platforms:
    - mega2560

compile:
  platforms:
    - mega2560
✔ ~/github/scls19fr/gokart_multilaptimer [master|…3]
10:40 $ bundle exec arduino_ci_remote.rb
Located Arduino binary...                                /Users/scls/Arduino.app
The set of compilers (1) isn't empty...                                        ✓
Checking g++ version...
    Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
    Apple LLVM version 10.0.0 (clang-1000.11.45.5)
    Target: x86_64-apple-darwin17.7.0
    Thread model: posix
    InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
...Checking g++ version                                                        ✓
libasan availability for g++...                                             true
Requested unittest platform 'mega2560' is defined in 'platforms' YML...        ✓
Unit testing test_laptimer.cpp with g++...

Last command:  $ g++ -std=c++0x -o /Users/scls/github/scls19fr/gokart_multilaptimer/unittest_test_laptimer.cpp.bin -DARDUINO=100 -g -O1 -fno-omit-frame-pointer -fno-optimize-sibling-calls -fsanitize=address -I/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino -I/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest -I/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Arduino.cpp /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Godmode.cpp /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/stdlib.cpp /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest/ArduinoUnitTests.cpp /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/utils.cpp /Users/scls/github/scls19fr/gokart_multilaptimer/test/test_laptimer.cpp

In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Arduino.cpp:1:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Godmode.h:2:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/ArduinoDefines.h:3:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/pgmspace.h:17:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/io.h:623:6: warning: "device type not defined" [-W#warnings]
#    warning "device type not defined"
     ^
1 warning generated.
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Godmode.cpp:1:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Godmode.h:2:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/ArduinoDefines.h:3:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/pgmspace.h:17:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/io.h:623:6: warning: "device type not defined" [-W#warnings]
#    warning "device type not defined"
     ^
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Godmode.cpp:3:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/SPI.h:97:11: error: use of undeclared identifier 'SPCR'
    if (!(SPCR & (1 << DORD))) {
          ^
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/SPI.h:97:24: error: use of undeclared identifier 'DORD'
    if (!(SPCR & (1 << DORD))) {
                       ^
1 warning and 2 errors generated.
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest/ArduinoUnitTests.cpp:1:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest/ArduinoUnitTests.h:3:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest/Assertion.h:7:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest/Compare.h:2:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/pgmspace.h:17:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/io.h:623:6: warning: "device type not defined" [-W#warnings]
#    warning "device type not defined"
     ^
1 warning generated.
In file included from /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:1:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Arduino.h:10:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/ArduinoDefines.h:3:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/pgmspace.h:17:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/io.h:623:6: warning: "device type not defined" [-W#warnings]
#    warning "device type not defined"
     ^
In file included from /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:1:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Arduino.h:17:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/SPI.h:97:11: error: use of undeclared identifier 'SPCR'
    if (!(SPCR & (1 << DORD))) {
          ^
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/SPI.h:97:24: error: use of undeclared identifier 'DORD'
    if (!(SPCR & (1 << DORD))) {
                       ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:24:5: error: use of undeclared identifier 'Serial'
    dbSerialPrintln("Too much sectors");
    ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:14:31: note: expanded from macro 'dbSerialPrintln'
  #define dbSerialPrintln(a)  dbSerial.println(a)
                              ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:10:18: note: expanded from macro 'dbSerial'
#define dbSerial Serial
                 ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:33:5: error: use of undeclared identifier 'Serial'
    dbSerialPrintln("Too big InitSector");
    ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:14:31: note: expanded from macro 'dbSerialPrintln'
  #define dbSerialPrintln(a)  dbSerial.println(a)
                              ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:10:18: note: expanded from macro 'dbSerial'
#define dbSerial Serial
                 ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:90:9: error: use of undeclared identifier 'Serial'
        dbSerial.print("            Last Lap : "); dbSerial.println(time_lastlap);
        ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:10:18: note: expanded from macro 'dbSerial'
#define dbSerial Serial
                 ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:90:52: error: use of undeclared identifier 'Serial'
        dbSerial.print("            Last Lap : "); dbSerial.println(time_lastlap);
                                                   ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:10:18: note: expanded from macro 'dbSerial'
#define dbSerial Serial
                 ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:93:9: error: use of undeclared identifier 'Serial'
        dbSerial.print("*********** Best Lap : "); dbSerial.println(time_bestlap);
        ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:10:18: note: expanded from macro 'dbSerial'
#define dbSerial Serial
                 ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:93:52: error: use of undeclared identifier 'Serial'
        dbSerial.print("*********** Best Lap : "); dbSerial.println(time_bestlap);
                                                   ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:10:18: note: expanded from macro 'dbSerial'
#define dbSerial Serial
                 ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp:96:7: error: use of undeclared identifier 'Serial'
      dbSerial.print("*********** first line");
      ^
/Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h:10:18: note: expanded from macro 'dbSerial'
#define dbSerial Serial
                 ^
1 warning and 9 errors generated.
In file included from /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/utils.cpp:1:
In file included from /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/utils.h:4:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Arduino.h:10:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/ArduinoDefines.h:3:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/pgmspace.h:17:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/io.h:623:6: warning: "device type not defined" [-W#warnings]
#    warning "device type not defined"
     ^
In file included from /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/utils.cpp:1:
In file included from /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/utils.h:4:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/Arduino.h:17:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/SPI.h:97:11: error: use of undeclared identifier 'SPCR'
    if (!(SPCR & (1 << DORD))) {
          ^
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/SPI.h:97:24: error: use of undeclared identifier 'DORD'
    if (!(SPCR & (1 << DORD))) {
                       ^
1 warning and 2 errors generated.
In file included from /Users/scls/github/scls19fr/gokart_multilaptimer/test/test_laptimer.cpp:1:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest/ArduinoUnitTests.h:3:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest/Assertion.h:7:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/unittest/Compare.h:2:
In file included from /Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/pgmspace.h:17:
/Users/scls/.rvm/gems/ruby-2.3.0/bundler/gems/arduino_ci-95857c830209/cpp/arduino/avr/io.h:623:6: warning: "device type not defined" [-W#warnings]
#    warning "device type not defined"
     ^
1 warning generated.
...Unit testing test_laptimer.cpp with g++                                     ✗
Indexing libraries...                                                          ✓
Installing library under test...                                               ✓
Library installed at... /Users/scls/Documents/Arduino/libraries/gokart_multilaptimer
Skipping builds; no examples found in /Users/scls/Documents/Arduino/libraries/gokart_multilaptimer...
    /Users/scls/github/scls19fr/gokart_multilaptimer/Gemfile
    /Users/scls/github/scls19fr/gokart_multilaptimer/Gemfile.lock
    /Users/scls/github/scls19fr/gokart_multilaptimer/README.md
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/config.h
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/debug.h
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.cpp
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/laptimer.h
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/mode.h
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/multilaptimer.ino
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/page.h
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/pinout.h
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/utils.cpp
    /Users/scls/github/scls19fr/gokart_multilaptimer/multilaptimer/utils.h
    /Users/scls/github/scls19fr/gokart_multilaptimer/test/test_laptimer.cpp
    /Users/scls/github/scls19fr/gokart_multilaptimer/unittest_test_laptimer.cpp.bin.dSYM/Contents/Info.plist
    /Users/scls/github/scls19fr/gokart_multilaptimer/unittest_test_laptimer.cpp.bin.dSYM/Contents/Resources/DWARF/unittest_test_laptimer.cpp.bin
...Skipping builds; no examples found in /Users/scls/Documents/Arduino/libraries/gokart_multilaptimer
Failures: 1
```

This is very clear... it's failing but I don't understand why

    error: use of undeclared identifier 'Serial'

Opening `multilaptimer/multilaptimer.ino` in Arduino IDE, I can compile this code.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,10 +6,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 ### Added
-
 - Provide an `itoa` function. It is present in Arduino's runtime environment but not on most (all?) host systems because itoa is not a portable standard function.
 
 ### Changed
+- Simplified the use of `Array.each` with a return statement; it's now simply `Array.find`
 
 ### Deprecated
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Changed
 - Simplified the use of `Array.each` with a return statement; it's now simply `Array.find`
+- `autolocate!` for Arduino installations now raises `ArduinoInstallationError` if `force_install` fails
 
 ### Deprecated
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Changed
 - Simplified the use of `Array.each` with a return statement; it's now simply `Array.find`
 - `autolocate!` for Arduino installations now raises `ArduinoInstallationError` if `force_install` fails
+- Errors due to missing YAML are now named `ConfigurationError`
 
 ### Deprecated
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -18,6 +18,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Removed
 
 ### Fixed
+- Determining a working OSX launch command no longer breaks on non-English installations
 
 ### Security
 

--- a/REFERENCE.md
+++ b/REFERENCE.md
@@ -405,7 +405,8 @@ unittest(interrupt_attachment) {
   assertEqual(state->interrupt[7].mode, 3);
   detachInterrupt(7);
   assertFalse(state->interrupt[7].attached);
-}```
+}
+```
 
 
 ### SPI

--- a/lib/arduino_ci/arduino_downloader.rb
+++ b/lib/arduino_ci/arduino_downloader.rb
@@ -186,7 +186,7 @@ module ArduinoCI
       elsif File.exist? extracted_file
         install
       else
-        puts "Arduino force-install failed"
+        puts "Could not find extracted archive (tried #{extracted_file})"
       end
 
       File.exist? self.class.force_install_location

--- a/lib/arduino_ci/arduino_downloader.rb
+++ b/lib/arduino_ci/arduino_downloader.rb
@@ -32,13 +32,7 @@ module ArduinoCI
       # if it exists.  I'm not sure why we would have both, but if we did
       # a force install then let's make sure we actually use it.
       locations = [self.force_installed_executable, self.existing_executable]
-      locations.each do |loc|
-        next if loc.nil?
-        next unless File.exist? loc
-
-        return loc
-      end
-      nil
+      locations.find { |loc| !loc.nil? && File.exist?(loc) }
     end
 
     # The autolocated directory of the installation
@@ -49,13 +43,7 @@ module ArduinoCI
       # if it exists.  I'm not sure why we would have both, but if we did
       # a force install then let's make sure we actually use it.
       locations = [self.force_install_location, self.existing_installation]
-      locations.each do |loc|
-        next if loc.nil?
-        next unless File.exist? loc
-
-        return loc
-      end
-      nil
+      locations.find { |loc| !loc.nil? && File.exist?(loc) }
     end
 
     # The path to the directory of an existing installation, or nil

--- a/lib/arduino_ci/arduino_downloader_osx.rb
+++ b/lib/arduino_ci/arduino_downloader_osx.rb
@@ -27,21 +27,17 @@ module ArduinoCI
     # @param Array<string> a list of places to look
     # @return [string]
     def self.find_existing_arduino_dir(paths)
-      paths.each do |path|
-        return path if File.exist? path
-      end
-      nil
+      paths.find(&File.method(:exist?))
     end
 
     # An existing Arduino file in one of the given directories, or nil
     # @param Array<string> a list of places to look for the executable
     # @return [string]
     def self.find_existing_arduino_exe(paths)
-      paths.each do |path|
+      paths.find do |path|
         exe = File.join(path, "MacOS", "Arduino")
-        return exe if File.exist? exe
+        File.exist? exe
       end
-      nil
     end
 
     # The path to the directory of an existing installation, or nil

--- a/lib/arduino_ci/arduino_downloader_windows.rb
+++ b/lib/arduino_ci/arduino_downloader_windows.rb
@@ -81,10 +81,10 @@ module ArduinoCI
     # @return [string]
     def self.existing_executable
       arduino_reg = 'SOFTWARE\WOW6432Node\Arduino'
-      Win32::Registry::HKEY_LOCAL_MACHINE.open(arduino_reg) do |reg|
+      Win32::Registry::HKEY_LOCAL_MACHINE.open(arduino_reg).find do |reg|
         path = reg.read_s('Install_Dir')
         exe = File.join(path, "arduino_debug.exe")
-        return exe if File.exist? exe
+        File.exist? exe
       end
     rescue
       nil
@@ -93,10 +93,7 @@ module ArduinoCI
     # The executable Arduino file in a forced installation, or nil
     # @return [string]
     def self.force_installed_executable
-      exe = File.join(self.force_install_location, "arduino_debug.exe")
-      return nil if exe.nil?
-
-      exe
+      File.join(self.force_install_location, "arduino_debug.exe")
     end
 
   end

--- a/lib/arduino_ci/arduino_installation.rb
+++ b/lib/arduino_ci/arduino_installation.rb
@@ -13,6 +13,8 @@ DESIRED_ARDUINO_IDE_VERSION = "1.8.6".freeze
 
 module ArduinoCI
 
+  class ArduinoInstallationError < StandardError; end
+
   # Manage the OS-specific install location of Arduino
   class ArduinoInstallation
 
@@ -94,7 +96,8 @@ module ArduinoCI
         return candidate unless candidate.nil?
 
         # force the install
-        force_install
+        raise ArduinoInstallationError, "Failed to force-install Arduino" unless force_install
+
         autolocate
       end
 

--- a/lib/arduino_ci/arduino_installation.rb
+++ b/lib/arduino_ci/arduino_installation.rb
@@ -80,7 +80,13 @@ module ArduinoCI
           # don't want to see is a java error.
           args = launcher + ["--bogus-option"]
           result = Host.run_and_capture(*args)
-          next unless result[:err].include? "Error: unknown option: --bogus-option"
+
+          # NOTE: Was originally searching for "Error: unknown option: --bogus-option"
+          #           but also need to find "Erreur: option inconnue : --bogus-option"
+          #           and who knows how many other languages.
+          # For now, just search for the end of the error and hope that the java-style
+          #  launch of this won't include a similar string in it
+          next unless result[:err].include? ": --bogus-option"
 
           ret.base_cmd = launcher
           ret.binary_path = Pathname.new(osx_root)

--- a/lib/arduino_ci/ci_config.rb
+++ b/lib/arduino_ci/ci_config.rb
@@ -35,6 +35,8 @@ UNITTEST_SCHEMA = {
 }.freeze
 module ArduinoCI
 
+  class ConfigurationError < StandardError; end
+
   # The filename controlling (overriding) the defaults for testing.
   # Files with this name can be used in the root directory of the Arduino library and in any/all of the example directories
   CONFIG_FILENAMES = [
@@ -108,7 +110,7 @@ module ArduinoCI
     # @return [ArduinoCI::CIConfig] a reference to self
     def load_yaml(path)
       yml = YAML.load_file(path)
-      raise "The YAML file at #{path} failed to load" unless yml
+      raise ConfigurationError, "The YAML file at #{path} failed to load" unless yml
 
       if yml.include?("packages")
         yml["packages"].each do |k, v|

